native/logikSimulation/src/data/component.rs

```diff
@@ -0,0 +1,35 @@
+/// A trait to define common behaviour between the components
+pub(crate) trait Component {
+    fn inputs(&self) -> usize;
+    fn outputs(&self) -> usize;
```

Comment:
I would like to see us do bi-directional ports on components so I think that should be considered in the design (But that can be down-prioritized).